tests/test_validation.py

```diff
@@ -698,6 +699,10 @@ def test_unfrozen_parameters_w_peft_layers_to_transform(self):
         ):
             validate_config(cfg)
 
+    def test_hub_model_id_save_value(self):
+        cfg = DictDefault({"hub_model_id": "test"})
+        validate_config(cfg)
```

Comment:
shouldn't this raise an error since neither save nor saves per epoch were set in this test? I think the logic may be busted somewhere
```suggestion
        with pytest.raises(
            ValueError,
            match=r".*set without any models being saved*",
        ):
            validate_config(cfg)```